### PR TITLE
refactor(dart_frog_prod): update to Dart 2.19 and Very Good Analysis 4.0.0

### PR DESCRIPTION
**Pull request by felangel**
## Status

**READY**

## Description

- refactor(dart_frog_prod): update to Dart 2.19 and Very Good Analysis 4.0.0

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [X] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: .github/workflows/dart_frog_prod_server.yaml [hunk 1]
```diff
@@ -14,8 +14,9 @@ on:
 
 jobs:
   build:
-    uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/dart_package.yml@v1
+    uses: ./.github/workflows/.dart_package.yaml
```
Comment (wolfenrain) [inline .github/workflows/dart_frog_prod_server.yaml:17]: Any reason that this is still needed?

Comment (felangel) [reply .github/workflows/dart_frog_prod_server.yaml:17]: No I need to revert and figure out why coverage is behaving differently in Linux vs MacOS

**Review by renancaraujo — APPROVED** [review]

File: .github/workflows/dart_frog_prod_server.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ jobs:
   build:
     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/dart_package.yml@v1
     with:
+      runs_on: macos-latest
```
Comment (wolfenrain) [inline .github/workflows/dart_frog_prod_server.yaml:19]: Did u figure out why the behavior was different?

**Review by wolfenrain — APPROVED** [review]

File: .github/workflows/dart_frog_prod_server.yaml (hunk 2 above)
Comment (felangel) [reply .github/workflows/dart_frog_prod_server.yaml:19]: Yeah, I was going to bring it up at standup but for now I wanted to unblock this so we can continue with the upgrades

**Pull request merged**